src/providers/seventv/SeventvEmotes.cpp

```diff
@@ -0,0 +1,361 @@
+                continue;
+            }
+            auto ptr = cachedOrMake(std::move(result.emote), result.id);
+            emotes[result.name] = ptr;
+        }
+
+        return emotes;
+    }
+
+}  // namespace
+
+SeventvEmotes::SeventvEmotes()
+    : global_(std::make_shared<EmoteMap>())
+{
+}
+
+std::shared_ptr<const EmoteMap> SeventvEmotes::globalEmotes() const
+{
+    return this->global_.get();
+}
+
+boost::optional<EmotePtr> SeventvEmotes::globalEmote(
+    const EmoteName &name) const
+{
+    auto emotes = this->global_.get();
+    auto it = emotes->find(name);
+
+    if (it == emotes->end())
+    {
+        return boost::none;
+    }
+    return it->second;
+}
+
+void SeventvEmotes::loadGlobalEmotes()
+{
+    if (!Settings::instance().enableSevenTVGlobalEmotes)
+    {
+        this->global_.set(EMPTY_EMOTE_MAP);
+        return;
+    }
+
+    qCDebug(chatterinoSeventv) << "Loading 7TV Global Emotes";
+
+    NetworkRequest(API_URL_GLOBAL_EMOTE_SET, NetworkRequestType::Get)
+        .timeout(30000)
+        .onSuccess([this](const NetworkResult &result) -> Outcome {
+            QJsonArray parsedEmotes = result.parseJson()["emotes"].toArray();
+
+            auto emoteMap = parseEmotes(parsedEmotes, true);
+            qCDebug(chatterinoSeventv)
+                << "Loaded" << emoteMap.size() << "7TV Global Emotes";
+            this->global_.set(std::make_shared<EmoteMap>(std::move(emoteMap)));
+
+            return Success;
+        })
+        .onError([](const NetworkResult &result) {
+            qCWarning(chatterinoSeventv)
+                << "Couldn't load 7TV global emotes" << result.getData();
+        })
+        .execute();
+}
+
+void SeventvEmotes::loadChannelEmotes(const std::weak_ptr<Channel> &channel,
+                                      const QString &channelId,
+                                      std::function<void(EmoteMap &&)> callback,
+                                      bool manualRefresh)
+{
+    qCDebug(chatterinoSeventv)
+        << "Reloading 7TV Channel Emotes" << channelId << manualRefresh;
+
+    NetworkRequest(API_URL_USER.arg(channelId), NetworkRequestType::Get)
+        .timeout(20000)
+        .onSuccess([callback = std::move(callback), channel, channelId,
+                    manualRefresh](const NetworkResult &result) -> Outcome {
+            auto json = result.parseJson();
+            auto emoteSet = json["emote_set"].toObject();
+            auto parsedEmotes = emoteSet["emotes"].toArray();
+
+            auto emoteMap = parseEmotes(parsedEmotes, false);
+            bool hasEmotes = !emoteMap.empty();
+
+            qCDebug(chatterinoSeventv)
+                << "Loaded" << emoteMap.size() << "7TV Channel Emotes for"
+                << channelId << "manual refresh:" << manualRefresh;
+
+            if (hasEmotes)
+            {
+                callback(std::move(emoteMap));
+            }
+            if (auto shared = channel.lock(); manualRefresh)
```

Comment:
I don't like this usage of `if (initialization; check)` where the initialization & check are completely different

I'd rather check manualRefresh first, then lock channel and output something if the lock succeeded